converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Modules/DicomModule.cs

```diff
@@ -32,15 +32,13 @@ public void Load(IServiceCollection services)
     {
         EnsureArg.IsNotNull(services, nameof(services));
 
-        var dicomWebConfiguration = new DicomWebConfiguration();
         IConfigurationSection dicomWebConfigurationSection = _configuration.GetSection(DicomWebConfigurationSectionName);
-        dicomWebConfigurationSection.Bind(dicomWebConfiguration);
+        services.AddOptions<DicomWebConfiguration>().Bind(dicomWebConfigurationSection);
 
-        services.AddSingleton(Options.Create(dicomWebConfiguration));
-
-        services.AddHttpClient<IDicomWebClient, DicomWebClient>(sp =>
+        services.AddHttpClient<IDicomWebClient, DicomWebClient>((sp, client) =>
             {
-                sp.BaseAddress = dicomWebConfiguration.Endpoint;
+                DicomWebConfiguration config = sp.GetRequiredService<IOptions<DicomWebConfiguration>>().Value;
+                client.BaseAddress = config.PrivateEndpoint == null ? config.Endpoint : config.PrivateEndpoint;
```

Comment:
Oh you know, we could have written this as `config.PrivateEndpoint ?? config.Endpoint` lol. But that doesn't matter